python_modules/dagster/dagster/_core/definitions/target.py

```diff
@@ -1,31 +1,25 @@
-from typing import TYPE_CHECKING, NamedTuple, Optional, Sequence, Union
+from typing import NamedTuple, Optional, Sequence, Union
 
 from typing_extensions import TypeAlias
 
 import dagster._check as check
 
+from .graph_definition import GraphDefinition
 from .job_definition import JobDefinition
 from .unresolved_asset_job_definition import UnresolvedAssetJobDefinition
 
-if TYPE_CHECKING:
-    from .graph_definition import GraphDefinition
-
 ExecutableDefinition: TypeAlias = Union[
     JobDefinition, "GraphDefinition", UnresolvedAssetJobDefinition
 ]
 
 
-class RepoRelativeTarget(NamedTuple):
-    """The thing to be executed by a schedule or sensor, selecting by name a job in the same repository."""
-
-    job_name: str
-    op_selection: Optional[Sequence[str]]
-
-
-class DirectTarget(
+class AutomationTarget(
     NamedTuple(
-        "_DirectTarget",
-        [("target", ExecutableDefinition)],
+        "_AutomationTarget",
+        [
+            ("job_resolvable", Union[ExecutableDefinition, str]),
+            ("op_selection", Optional[Sequence[str]]),
+        ],
     )
 ):
     """The thing to be executed by a schedule or sensor, referenced directly and loaded
```

Comment:
Since we are cleaning this up I think we should write a coherent docblock here.

In the case where an automation targets a selection of assets, do we make some sort of anonymous job? What is this when it is a string?

Reply:
I will add a clear docblock.

>In the case where an automation targets a selection of assets, do we make some sort of anonymous job?

We do not, which is one major source of the weirdness in `SensorDefinition`. When an `asset_selection` is passed, there is no "target" at all. I didn't change that in my stack, but I'd like to investigate it later (in particular after enabling multi-partitions-def asset jobs, since that change will make any anonymous job logic simpler).

>What is this when it is a string?

When it's a string it is a job name (which was previously stored on `RepoRelativeTarget`).